REGRESSION_TESTS.md

```diff
@@ -0,0 +1,41 @@
+This document assumes you have already been able to build CPROVER on
+your chosen architecture. For details on building CPROVER see
+COMPILING.md.
```

Comment:
Could we please merge this documentation with doc/architectural/compilation-and-development.md rather than adding another file?

Reply:
Happy to merge this file with another one. I started a draft pull request to make sure I wasn't putting bad information in or adding things in the wrong place. If we all agree on doc/architectural/compilation-and-development.md as the right location I'll move everything there.

Reply:
Thank you for bringing up all these gaps in our documentation, and coming up with fixes for these! I'd just like to make sure we don't have the documentation in multiple places. I'm no longer able to tell what's actually needed for someone to get started, and where they might be looking for documentation. Is perhaps `CONTRIBUTING.md` (which we don't currently have) the canonical place for people to get started?

Reply:
I'm happy to try and sort it all out. I'm using myself as a test subject for this (before I know my way around well), so hopefully do not create too many (draft) proposals to change things that are explained somewhere.

For (a possible) `CONTRIBUTING.md`, would this also include the various links and information on policies for style and committing?

Reply:
If we had a `CONTRIBUTING.md`, perhaps it could serve as a collection of pointers towards documentation, pointing people to `COMPILING.md`, `CODING_STANDARD.md`, and the sections of the manual (such as `doc/architectural/compilation-and-development.md`).

Reply:
Btw, the documentation is published at http://cprover.diffblue.com (see link on top fo README)

Reply:
What is our link preference here, should we link to http://cprover.diffblue.com/ or do we reference inside the repository since that page is generated from doc/architectural/front-page.md ?

Reply:
`README.md` references http://cprover.diffblue.com/, so I'd go with that in top-level files (`README.md`, `COMPILING.md`, `CODING_STANDARD.md`, and possibly any files to be added at that level).